module/product-collection/src/Infrastructure/Validator/ProductCollectionTypeExistsValidator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\ProductCollection\Infrastructure\Validator;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Ergonode\ProductCollection\Domain\Repository\ProductCollectionTypeRepositoryInterface;
+use Ergonode\SharedKernel\Domain\Aggregate\ProductCollectionTypeId;
+use Ergonode\ProductCollection\Infrastructure\Validator\Constraints\ProductCollectionTypeExists;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+class ProductCollectionTypeExistsValidator extends ConstraintValidator
+{
+    private ProductCollectionTypeRepositoryInterface $repository;
+
+    public function __construct(ProductCollectionTypeRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @param mixed                                  $value
+     * @param ProductCollectionTypeExists|Constraint $constraint
+     */
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof ProductCollectionTypeExists) {
+            throw new UnexpectedTypeException($constraint, ProductCollectionTypeExists::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!is_scalar($value) && !(\is_object($value) && method_exists($value, '__toString'))) {
+            throw new UnexpectedTypeException($value, 'string');
+        }
+
+        $value = (string) $value;
+
+        if (!ProductCollectionTypeId::isValid($value)) {
```

Comment:
I think not valid id should lead to violation - one thing is that potential value might be empty but when the id is given but is not valid it is confusing that the validator does not rise violation